### ISSUE TITLE
D1X-ReBirth v0.60 BETA 2: Crash when someone joins a in-progress "Open" game.

### ISSUE THREAD

**Issue by KynikossDragonn** [dxx-rebirth/dxx-rebirth#402]
### Environment

Compiled from: https://www.dxx-rebirth.com/download/dxx/user/afuturepilot/dxx-rebirth-0.60.0-beta2-git.tar.gz

#### Operating System Environment

* Linux Mint
  * [x] Mint 17.3 (Rosa)

#### CPU environment

* [x] x86\_64 (64-bit Intel/AMD)

#### Game environment
Descent: First Strike
Total Chaos

Descent v1.4a

### Description

Multiplayer games can start just fine from the "Forming" stage once all clients connected, but if the game is in "Open" mode and someone tries to connect to the game after it has launched into gameplay it crashes with the following:

```
terminate called after throwing an instance of 'std::length_error'
  what():  too many blown bitmaps
Aborted
```

### Steps to Reproduce

#1 - Host a multiplayer game.
#2 - Set game mode to "Open".
#3 - Launch game. (with or without players already in the forming stage)
#4 - Let someone connect after game launch.

Observed behaviour:
The host as well as everyone else in the game currently will crash with the abort mentioned above.

Expected behaviour:
The additional client spawns in just fine for everybody else and game continues as normal.

* Engines affected:
  * [x] D1X-Rebirth
  * [ ] D2X-Rebirth

* Game modes affected:
  * [ ] Single player
  * [x] Multiplayer competitive
    * [x] Anarchy
    * [ ] Team Anarchy
    * [ ] Robo-Anarchy
    * [ ] Capture the Flag
    * [ ] Bounty
  * [x] Multiplayer cooperative

* Players affected by issue:
  * [x] Game host
  * [x] Game guests

**Comment by vLKp** [comment]
This appears to be a duplicate of [[d1x windows 10] Game crashes when my hosted game is joined](https://forum.dxx-rebirth.com/showthread.php?tid=1053).  At the end of that thread, I committed and published a potential fix ([Improve error handling/reporting for excessive blown bitmaps](/dxx-rebirth/dxx-rebirth/commit/b1ba7c27d176ac145fb8666b30db61c29b694531)), but no one ever confirmed whether it resolved the crash.  Please try cherry-picking that patch into your build and report back.

**Comment by KynikossDragonn** [comment]
Github is being really unresponsive in my browser for some reason, is this commit just net_udp the only file changed that I should cherry pick over?

If so, I'll get this compiled and report back as soon as I have someone available to test out with me again.

**Comment by vLKp** [comment]
That is the only changed file, but there are other unrelated changes to that file between [0.60.0-beta2](/dxx-rebirth/dxx-rebirth/commit/8a16296eb4aa50ff086eacd7b30cff7a81be456f) and [the commit I suggested](/dxx-rebirth/dxx-rebirth/commit/b1ba7c27d176ac145fb8666b30db61c29b694531), so copying the file whole from that commit will not work, since some of the other changes you would pick up require changes to other files.  If you have a local git checkout, you can use `git cherry-pick b1ba7c27d176ac145fb8666b30db61c29b694531` to bring in just the fix.  If you want to get the change over the web interface, [append `.patch` to the URL](https://github.com/dxx-rebirth/dxx-rebirth/commit/b1ba7c27d176ac145fb8666b30db61c29b694531.patch).

On the assumption that this comment might be e-mailed to you by the system (thereby bypassing any browser issues you have with Github), I have included below a base64 encoded copy of the diff.  To use it, run the text through `base64 -d` to produce a `.patch` file, then use `patch -p1` to apply it from the top level of the unpacked source tree.

```
ZGlmZiAtLWdpdCBhL3NpbWlsYXIvbWFpbi9uZXRfdWRwLmNwcCBiL3NpbWlsYXIvbWFpbi9uZXRf
dWRwLmNwcAppbmRleCBmMzk0MzdlZjIuLjEzMmVhOWEyMSAxMDA2NDQKLS0tIGEvc2ltaWxhci9t
YWluL25ldF91ZHAuY3BwCisrKyBiL3NpbWlsYXIvbWFpbi9uZXRfdWRwLmNwcApAQCAtMTgyOSwx
MyArMTgyOSw4IEBAIG5hbWVzcGFjZSB7CiAKIGNsYXNzIGJsb3duX2JpdG1hcF9hcnJheQogewot
I2lmIGRlZmluZWQoRFhYX0JVSUxEX0RFU0NFTlRfSSkKLSNkZWZpbmUgTlVNX0JMT1dOX0JJVE1B
UFMgNwotI2VsaWYgZGVmaW5lZChEWFhfQlVJTERfREVTQ0VOVF9JSSkKLSNkZWZpbmUgTlVNX0JM
T1dOX0JJVE1BUFMgMjAKLSNlbmRpZgogCXR5cGVkZWYgaW50IFQ7Ci0JdHlwZWRlZiBhcnJheTxU
LCBOVU1fQkxPV05fQklUTUFQUz4gYXJyYXlfdDsKKwl1c2luZyBhcnJheV90ID0gYXJyYXk8VCwg
MzI+OwogCXR5cGVkZWYgYXJyYXlfdDo6Y29uc3RfaXRlcmF0b3IgY29uc3RfaXRlcmF0b3I7CiAJ
YXJyYXlfdCBhOwogCWFycmF5X3Q6Oml0ZXJhdG9yIGU7CkBAIC0xODU0LDcgKzE4NDksMTAgQEAg
cHVibGljOgogCQlpZiAoZXhpc3RzKHQpKQogCQkJcmV0dXJuOwogCQlpZiAoZSA9PSBhLmVuZCgp
KQotCQkJdGhyb3cgc3RkOjpsZW5ndGhfZXJyb3IoInRvbyBtYW55IGJsb3duIGJpdG1hcHMiKTsK
KwkJeworCQkJTGV2ZWxFcnJvcigidG9vIG1hbnkgYmxvd24gYml0bWFwczsgaWdub3JpbmcgYml0
bWFwICVpLiIsIHQpOworCQkJcmV0dXJuOworCQl9CiAJCSplID0gdDsKIAkJKytlOwogCX0KQEAg
LTE4NjIsNDEgKzE4NjAsNDUgQEAgcHVibGljOgogCiB9CiAKLXN0YXRpYyBpbnQgbmV0X3VkcF9j
cmVhdGVfbW9uaXRvcl92ZWN0b3Iodm9pZCkKK3N0YXRpYyB1bnNpZ25lZCBuZXRfdWRwX2NyZWF0
ZV9tb25pdG9yX3ZlY3Rvcih2b2lkKQogewotCWludCBtb25pdG9yX251bSA9IDA7Ci0JaW50IHZl
Y3RvciA9IDA7CiAJYmxvd25fYml0bWFwX2FycmF5IGJsb3duX2JpdG1hcHM7CisJY29uc3RleHBy
IHNpemVfdCBtYXhfdGV4dHVyZXMgPSBUZXh0dXJlcy5zaXplKCk7CiAJcmFuZ2VfZm9yIChhdXRv
ICZpLCBwYXJ0aWFsX2NvbnN0X3JhbmdlKEVmZmVjdHMsIE51bV9lZmZlY3RzKSkKIAl7Ci0JCWlm
IChpLmRlc3RfYm1fbnVtIDwgVGV4dHVyZXMuc2l6ZSgpKQorCQlpZiAoaS5kZXN0X2JtX251bSA8
IG1heF90ZXh0dXJlcykKIAkJewogCQkJYmxvd25fYml0bWFwcy5pbnNlcnRfdW5pcXVlKGkuZGVz
dF9ibV9udW0pOwogCQl9CiAJfQotCQkKLQlyYW5nZV9mb3IgKGNvbnN0IGF1dG8gJiZzZWcsIHZj
c2VncHRyKQorCXVuc2lnbmVkIG1vbml0b3JfbnVtID0gMDsKKwl1bnNpZ25lZCB2ZWN0b3IgPSAw
OworCXJhbmdlX2ZvciAoY29uc3QgYXV0byAmJnNlZywgdmNzZWdwdHJpZHgpCiAJewotCQlpbnQg
dG0sIGVjOwogCQlyYW5nZV9mb3IgKGF1dG8gJmosIHNlZy0+c2lkZXMpCiAJCXsKLQkJCWlmICgo
dG0gPSBqLnRtYXBfbnVtMikgIT0gMCkKKwkJCWNvbnN0IHVuc2lnbmVkIHRtMiA9IGoudG1hcF9u
dW0yOworCQkJaWYgKCF0bTIpCisJCQkJY29udGludWU7CisJCQljb25zdCB1bnNpZ25lZCBtYXNr
ZWRfdG0yID0gdG0yICYgMHgzZmZmOworCQkJY29uc3QgdW5zaWduZWQgZWMgPSBUbWFwSW5mb1tt
YXNrZWRfdG0yXS5lY2xpcF9udW07CiAJCQl7Ci0JCQkJaWYgKChlYyA9IFRtYXBJbmZvW3RtICYg
MHgzZmZmXS5lY2xpcF9udW0pICE9IGVjbGlwX25vbmUgJiYKKwkJCQlpZiAoZWMgIT0gZWNsaXBf
bm9uZSAmJgogCQkJCQlFZmZlY3RzW2VjXS5kZXN0X2JtX251bSAhPSB+MHUpCiAJCQkJewotCQkJ
CQltb25pdG9yX251bSsrOwotCQkJCQlBc3NlcnQobW9uaXRvcl9udW0gPCAzMik7CiAJCQkJfQot
CQkJCWVsc2UKKwkJCQllbHNlIGlmIChibG93bl9iaXRtYXBzLmV4aXN0cyhtYXNrZWRfdG0yKSkK
IAkJCQl7Ci0JCQkJCWlmIChibG93bl9iaXRtYXBzLmV4aXN0cyh0bSYweDNmZmYpKQorCQkJCQlp
ZiAobW9uaXRvcl9udW0gPj0gOCAqIHNpemVvZih2ZWN0b3IpKQogCQkJCQl7Ci0JCQkJCQkJdmVj
dG9yIHw9ICgxIDw8IG1vbml0b3JfbnVtKTsKLQkJCQkJCQltb25pdG9yX251bSsrOwotCQkJCQkJ
CUFzc2VydChtb25pdG9yX251bSA8IDMyKTsKKwkJCQkJCUxldmVsRXJyb3IoInRvbyBtYW55IGJs
b3duIG1vbml0b3JzOyBpZ25vcmluZyBzZWdtZW50ICVodS4iLCBzZWcuZ2V0X3VuY2hlY2tlZF9p
bmRleCgpKTsKKwkJCQkJCXJldHVybiB2ZWN0b3I7CiAJCQkJCX0KKwkJCQkJCQl2ZWN0b3IgfD0g
KDEgPDwgbW9uaXRvcl9udW0pOwogCQkJCX0KKwkJCQllbHNlCisJCQkJCWNvbnRpbnVlOworCQkJ
CW1vbml0b3JfbnVtKys7CiAJCQl9CiAJCX0KIAl9Cg==
```

**Comment by KynikossDragonn** [comment]
Yeah, I got the file properly patched. Just waiting on some friends of mine to become available to test.

**Comment by vLKp** [comment]
The reported issue is long since patched.  Closing.

Please reopen if the issue persists for you.
